neoemf-core/src/main/java/fr/inria/atlanmod/neoemf/core/AbstractIdProvider.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2013-2018 Atlanmod, Inria, LS2N, and IMT Nantes.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v2.0 which accompanies
+ * this distribution, and is available at https://www.eclipse.org/legal/epl-2.0/
+ */
+
+package fr.inria.atlanmod.neoemf.core;
+
+import fr.inria.atlanmod.commons.hash.Hasher;
+import fr.inria.atlanmod.commons.hash.StandardHashers;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+/**
+ * An abstract {@link IdProvider}.
+ */
+@ParametersAreNonnullByDefault
+public abstract class AbstractIdProvider implements IdProvider {
+
+    /**
+     * The hasher used to generate {@link Id} from string.
+     */
+    @Nonnull
+    private final Hasher hasher = StandardHashers.XX;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this final field static too. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1170)